compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -501,6 +511,74 @@ private void addTypeId(MethodVisitor mv, BTypeIdSet.BTypeId typeId, boolean isPr
         return targetLabels;
     }
 
+    // -------------------------------------------------------
+    //              getAnonType() generation methods
+    // -------------------------------------------------------
+
+    void generateGetAnonTypeMethod(ClassWriter cw, List<BIRTypeDefinition> typeDefinitions, String typeOwnerClass) {
+        MethodVisitor mv = cw.visitMethod(ACC_PUBLIC, GET_ANON_TYPE,
+                String.format("(IL%s;)L%s;", STRING_VALUE, TYPE), null, null);
+        mv.visitCode();
+
+        int hashParamRegIndex = 1;
+        int shapeParamRegIndex = 2;
+        Label defaultCaseLabel = new Label();
+
+        // filter anon types and sorts them before generating switch case.
+        Set<BIRTypeDefinition> typeDefSet = new TreeSet<>(TYPE_HASH_COMPARATOR);
+        for (BIRTypeDefinition t : typeDefinitions) {
+            if (t.internalName.value.contains(BLangAnonymousModelHelper.ANON_PREFIX)) {
```

Comment:
I think there are some cases where we generate types/names but they don't have the anon prefix. Shall we create an issue to track such types and maybe use the ANONYMOUS flag here?

Reply:
created: https://github.com/ballerina-platform/ballerina-lang/issues/29948

Reply:
Fixed in: 3da69b3